### PR TITLE
Add-on Store: add sub menu items and various improvements

### LINKED ISSUE #2047: The addon store sub-categories
The new [addon store](https://github.com/openhab/openhab-webui/pull/1997) link on the left looks nice!

One suggestion is to add the sub-sections on the left with:
- Bindings
- Automation
- UI 
- Other (or should we expand this to)
- System Integrations
- Persistence
- Transformation
- Voice & Speech

### PR DESCRIPTION
**Pull request by jimtng**
Resolve #2047

The subsections are implemented as routable tabs, so switching around the different sections is fast.

image

## Installed addons are grouped by category in the main Add-on Store section
image

## Search in the main "Add-on Store" menu will search all types of addons
image

## Search in the corresponding section will only search for add-ons for that category
image

image

### CODE REVIEW

**Comment by jimtng** [comment]
What I couldn't figure out: How to push the visits to the sub-sections into the browser's history. Framework7's documentation seems to state that using routable tabs would make that happen automatically, but it doesn't. I tried using `history.pushState()` but the resulting behaviour is weird, so I took it out.

**Comment by jimtng** [comment]
@florian-h05 what's the objection for caching the add-ons data? It would get updated anyway so staleness is very minimal. The benefit is a more responsive experience and less twirling circle waiting

**Comment by florian-h05** [comment]
I have already written a command but not submitted review yet, so it is not showing app ...

The objection against this way of "caching" is that IMHO caching should be properly implemented on the server side.

**Comment by jimtng** [comment]
> caching should be properly implemented on the server side.

Is there currently a general implementation of mechanism in core for this?

**Review by florian-h05 — APPROVED** [review]
Overall LGTM, I have commited some changes.

File: bundles/org.openhab.ui/web/src/components/app.vue [hunk 1]
```diff
@@ -280,6 +291,9 @@ export default {
     }
 
     return {
+      AddonIcons: AddonIcons,
```
[the remark this answers is not in the document]
Comment (florian-h05) [reply bundles/org.openhab.ui/web/src/components/app.vue:294]: Sure, that's a point to use `created`. I will revert that change.

**Comment by florian-h05** [comment]
> Is there currently a general implementation or mechanism in core for this?

As the REST endpoints have different data sources, there is no general implementation.
Usually, RegistryChangeListeners are used to listen for data changes, but in some cases other ways of "watching" for changes need to be used.
I am currently working on a caching implementation for the add-ons endpoint.

**Comment by florian-h05** [comment]
@jimtng Ready to merge?

**Comment by jimtng** [comment]
Thanks for the review and fixes. I wish the caching stayed in but hopefully the server side implementation you're working on will work as well.

**Comment by jimtng** [comment]
> @jimtng Ready to merge?

Yes thanks

**Comment by ghys** [comment]
To be honest I wasn't sure of the ability to navigate the add-on store on narrow screens but aftter all I think it's okay - for now:

image

The bottom toolbar (which I thought initially to be removed) is nice but is becoming quite crowded.
This is why I made the add-on store initially about 3 main concepts: bindings, automation, UI (and the rest in a 4th tab, and universal search in a 5th tab).

**Comment by florian-h05** [comment]
I added the toolbar for those devices where the sidebar is not permanently available and have also noticed that it is crowded.
Having one icon less in the tab bar would already help with the space.

**Comment by jimtng** [comment]
Tbh I've never tried accessing the main UI from a mobile phone, but now I've just tried it.

I've noticed several things:
- Because the focus goes to the search bar every time I switched to a new tab, it's annoying that the keyboard would pop up an converting up half the screen
- The suggested and recommended items seem too big and take up a lot of space (same in the old design)

- I'm not 100% happy with the main "add-ons" tab showing only the installed add-ons. I'd like each section to have a little link or button for "more" that goes to the corresponding tab. But that may be redundant with the bottom tab bar?

**Comment by ghys** [comment]
> Tbh I've never tried accessing the main UI from a mobile phone

This is indeed a problem and something that you MUST do before altering designs.
You just have to test your changes with the 3 themes, both dark & light, and on various devices before you submit impactful changes.
We have 3 seats available on our sponsored BrowserStack team and I'm happy to give one to you as you seem to be serious about developing the UI. Just reach out with an email address in private on the forum.

File: bundles/org.openhab.ui/web/src/pages/addons/addons-store.vue [hunk 2]
```diff
@@ -95,10 +119,11 @@
           @addonButtonClick="addonButtonClick"
           :addons="allAddons.filter((a) => a.type === 'automation' && a.contentType === 'application/vnd.openhab.uicomponent;type=blocks')"
           :install-action-text="'Add'"
-          :title="'Block Libraries'"
+          :title="'Blockly Libraries'"
```
Comment (ghys) [inline bundles/org.openhab.ui/web/src/pages/addons/addons-store.vue:122]: These are called "Block Libraries" and have been from the start.
Sorry but you don't get to rename them on a whim.

Comment (ghys) [reply bundles/org.openhab.ui/web/src/pages/addons/addons-store.vue:122]: #2404

Comment (florian-h05) [reply bundles/org.openhab.ui/web/src/pages/addons/addons-store.vue:122]: Thanks for the catch and the fix!

**Pull request merged**